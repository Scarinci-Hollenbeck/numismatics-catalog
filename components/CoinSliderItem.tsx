import React from 'react';
import Link from 'next/link';
import CoinImage from './CoinImage';
import ButtonLink from './ButtonLink';
import { makeUrl } from '../utils/helpers';
import { Slide } from '../interfaces';

type Props = {
  slide: Slide
}

export default function CoinSliderItem({ slide }: Props): JSX.Element {
  return (
    <figure>
      <CoinImage image={slide.imageUrl} title={slide.title} />
      <figcaption>
        <h3>{slide.title}</h3>
        <p>{slide.category}</p>
        <ButtonLink
          link={`/coin/${encodeURIComponent(slide._id)}/${encodeURIComponent(
            makeUrl(slide.title),
          )}`}
          caption="Details"
        />
      </figcaption>

      <style jsx>
        {`
          figure {
            font-family: 'Tajawal Regular';
            font-size: 1.3rem;
            display: flex;
            flex-direction: column;
          }

          figure h3,
          figure h5,
          figure figcaption {
            margin: 0;
          }

          figure h3 {
            font-size: 1.5rem;
          }
        `}
      </style>
    </figure>
  );
}
