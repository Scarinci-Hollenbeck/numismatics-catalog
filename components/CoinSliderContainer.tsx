import React from 'react';
import useSWR from 'swr';
import Carousel from 'react-multi-carousel';
import CoinSliderItem from './CoinSliderItem';
import { getFetcher } from '../utils/helpers';
import { Slide } from '../interfaces';

type Props = {
  deviceType: string
}

export default function CoinSliderContainer({
  deviceType,
}: Props): JSX.Element {
  const { data: slides, error: slidesError } = useSWR(
    '/api/list-latest-coins',
    getFetcher,
  );

  const responsive = {
    desktop: {
      breakpoint: { max: 3000, min: 1024 },
      items: 1,
      partialVisibilityGutter: 60,
    },
    tablet: {
      breakpoint: { max: 1024, min: 464 },
      items: 1,
      partialVisibilityGutter: 50,
    },
    mobile: {
      breakpoint: { max: 464, min: 0 },
      items: 1,
      partialVisibilityGutter: 30,
    },
  };

  return (
    <section className="coin-slider">
      <h2>Newly added coins</h2>
      <hr />
      {slides !== undefined && slides.data.length > 0 && (
        <Carousel
          additionalTransfrom={0}
          arrows
          draggable
          focusOnSelect={false}
          infinite
          keyBoardControl
          minimumTouchDrag={80}
          renderButtonGroupOutside={false}
          renderDotsOutside={false}
          responsive={responsive}
          deviceType={deviceType}
          containerClass="carousel-container"
        >
          {slides.data.map((slide: Slide) => (
            <CoinSliderItem key={slide._id} slide={slide} />
          ))}
        </Carousel>
      )}
      <style jsx>
        {`
          section {
            background-color: #fff;
            box-shadow: 2px 4px 20px #a9a9a9;
            max-width: 100%;
            padding: 1em;
          }

          section h2 {
            font-family: 'Tajawal Regular';
            font-size: 2rem;
            margin: 0;
            padding: 0;
            text-align: left;
            color: #e74c3c;
          }

          section hr {
            background-color: #e74c3c;
            border: 0;
            height: 5px;
            border-radius: 5px;
          }
        `}
      </style>
    </section>
  );
}
