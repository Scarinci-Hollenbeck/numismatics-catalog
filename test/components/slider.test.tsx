import React from 'react'
import renderer from 'react-test-renderer'
import CoinSliderItem from '../../components/CoinSliderItem'
import CoinSliderContainer from '../../components/CoinSliderContainer'

const slides = [
  {
    id: '123456',
    title: 'King Ferdinand VI, 1746 - 1759',
    image: 'https://coincatalogds.nyc3.digitaloceanspaces.com/15-set.jpg',
    description:
      'Obv: Ferdin. VI D.G. Hisp. Et Ind. R. 1746-1759, Rev: Aeqvilibrivm Weight: 63.7864 Grams, 2.25 Ounces Measurements: .50 MM Diameter, 2 Inches In Diameter',
    link: 'king-ferdinand-vi-1746-1759',
  },
  {
    id: '123456',
    title: 'King Ferdinand VI, 1746 - 1759',
    image: 'https://coincatalogds.nyc3.digitaloceanspaces.com/15-set.jpg',
    description:
      'Obv: Ferdin. VI D.G. Hisp. Et Ind. R. 1746-1759, Rev: Aeqvilibrivm Weight: 63.7864 Grams, 2.25 Ounces Measurements: .50 MM Diameter, 2 Inches In Diameter',
    link: 'king-ferdinand-vi-1746-1759',
  },
  {
    id: '123456',
    title: 'King Ferdinand VI, 1746 - 1759',
    image: 'https://coincatalogds.nyc3.digitaloceanspaces.com/15-set.jpg',
    description:
      'Obv: Ferdin. VI D.G. Hisp. Et Ind. R. 1746-1759, Rev: Aeqvilibrivm Weight: 63.7864 Grams, 2.25 Ounces Measurements: .50 MM Diameter, 2 Inches In Diameter',
    link: 'king-ferdinand-vi-1746-1759',
  },
]

describe('Slider Item component', () => {
  it('matches snapshot', () => {
    const tree = renderer.create(<CoinSliderItem slide={slides[0]} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('Slider Container component', () => {
  it('matches snapshot', () => {
    const tree = renderer
      .create(<CoinSliderContainer deviceType="mobile" slides={slides} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
