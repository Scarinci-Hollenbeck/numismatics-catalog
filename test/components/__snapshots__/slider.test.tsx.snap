// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider Container component matches snapshot 1`] = `
<section
  className="coin-slider"
>
  <h2>
    Newly added coins
  </h2>
  <hr />
  <div
    className="react-multi-carousel-list carousel-container "
  >
    <ul
      className="react-multi-carousel-track "
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "overflow": "unset",
          "transform": "translate3d(0px,0,0)",
          "transition": "none",
        }
      }
    />
  </div>
</section>
`;

exports[`Slider Item component matches snapshot 1`] = `
<figure
  className="slide-item"
>
  <img
    alt="King Ferdinand VI, 1746 - 1759"
    src="https://coincatalogds.nyc3.digitaloceanspaces.com/15-set.jpg"
  />
  <figcaption>
    <h3>
      King Ferdinand VI, 1746 - 1759
    </h3>
    <h5>
      <a
        href="/coin/123456/king-ferdinand-vi-1746-1759"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Details &gt;&gt;
      </a>
    </h5>
  </figcaption>
</figure>
`;
